### PR TITLE
pack plugin script

### PR DESCRIPTION
**Pull request by keksa**

### CODE REVIEW

File: bin/pack-plugin [hunk 1]
```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env php
+<?php
+
+if (! class_exists(ZipArchive::class)) {
+    echo 'This script requires zip PHP extension.' . PHP_EOL;
```
Comment (Piskvor) [inline bin/pack-plugin:5]: composer suggest ext-zip?

File: bin/pack-plugin [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    if (! preg_match('~^[a-z0-9_-]+$~', $name)) {
+        echo 'Plugin\'s name contains invalid characters.' . PHP_EOL;
+
+        exit(1);
+    }
+
+    return $name;
+}
+
+$pluginRootDir = getPluginRootDir($argv[1] ?? null);
```
Comment (Piskvor) [inline bin/pack-plugin:69]: do we want to check if the plugin dir is sane, other than "has manifest"?

**Review by janprochazkacz — APPROVED** [review]

File: bin/pack-plugin [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    $name = $manifest['information']['name'] ?? null;
+    if ($name === null) {
+        echo 'Could not get plugin\'s name from manifest.' . PHP_EOL;
+
+        exit(1);
+    }
+
+    if (! preg_match('~^[a-z0-9_-]+$~', $name)) {
+        echo 'Plugin\'s name contains invalid characters.' . PHP_EOL;
```
Comment (Piskvor) [inline bin/pack-plugin]: No dots? No [A-Z]? Also, "Valid characters are: A-Z,a-z,0-9,_ (underscore), and - (hyphen)"?

File: bin/pack-plugin [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    exit(1);
+}
+
+$files = new CallbackFilterIterator(
+    new \RecursiveIteratorIterator(
+        new \RecursiveDirectoryIterator($pluginRootDir)
+    ),
+    function (SplFileInfo $fileInfo) {
+        return ! $fileInfo->isDir();
```
Comment (Piskvor) [inline bin/pack-plugin]: Hmm...symlinks? Perhaps `return $fileInfo->isFile()`?

File: bin/pack-plugin [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+// If the plugin's structure matches UBNT repository, create the zip next to the src directory.
+// Otherwise create it in plugin's root directory.
+if (file_exists(sprintf('%s/../README.md', $pluginRootDir)) && is_dir(sprintf('%s/../src', $pluginRootDir))) {
+    $zipPath = sprintf('%s/../%s', $pluginRootDir, $zipName);
+} else {
+    $zipPath = sprintf('%s/%s', $pluginRootDir, $zipName);
+}
+
+chdir($pluginRootDir);
```
Comment (Piskvor) [inline bin/pack-plugin]: `or die()`

**Review by Piskvor — CHANGES_REQUESTED** [review]

**Pull request merged**